test/freenet/clients/http/utils/UriFilterProxyHeaderParserTest.java

```diff
@@ -0,0 +1,95 @@
+package freenet.clients.http.utils;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import junit.framework.TestCase;
+
+import freenet.support.MultiValueTable;
+
+@RunWith(Parameterized.class)
+public class UriFilterProxyHeaderParserTest extends TestCase {
+
+  private final String fProxyPort;
+  private final String fProxyBindTo;
+  private final String uriScheme;
+  private final String uriHost;
+  private final MultiValueTable<String, String> headers;
+  private String resultUriPrefix;
+
+  @Parameterized.Parameters(
+      name = "{index}: portConfig=\"{0}\", bindTo=\"{1}\", uriScheme=\"{2}\", uriHost=\"{3}\", headers={4}, expected=\"{5}\"")
+  public static Collection constructorParametersAndResult () {
+    return Arrays.asList(new Object[][] {
```

Comment:
I am not a big fan of these parameterized tests because if test 17 of those fails I have _no_ idea whatsover how that test differs from the other ones, i.e. what makes _this_ test so special that it fails… I’d prefer one method per actual test, e.g. “testThatInvalidHostInHeaderIsReplacedByFproxyHost” and similarly-named tests.

Reply:
In this case (due to name = "{index}: …"), you get information how the data in the test changes. But I can change this with a moderate amount of work. Watch me crank out Emacs macros :-)